compiler/aarch64/codegen/FPTreeEvaluator.cpp

```diff
@@ -818,12 +818,60 @@ OMR::ARM64::TreeEvaluator::dcmpleEvaluator(TR::Node *node, TR::CodeGenerator *cg
    return fcmpHelper(node, TR::CC_LS, true, cg);
    }
 
+static TR::Register *
+floatThreeWayCompareHelper(TR::Node *node, bool isDouble, bool isCmpl, TR::CodeGenerator *cg)
+   {
+   // dcmpl/fcmpl: 1 if v1 > v2, 0 if v1 == v2, -1 if v1 < v2 or unordered
+   // dcmpg/fcmpg: 1 if v1 > v2 or unordered, 0 if v1 == v2, -1 if v1 < v2
+
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Node *secondChild = node->getSecondChild();
+   TR::Register *src1Reg = cg->evaluate(firstChild);
+   TR::Register *src2Reg = cg->evaluate(secondChild);
+   TR::Register *trgReg = cg->allocateRegister();
+   TR::LabelSymbol *doneLabel = generateLabelSymbol(cg);
+   TR::InstOpCode::Mnemonic cmpOp = isDouble ? TR::InstOpCode::fcmpd : TR::InstOpCode::fcmps;
+   TR::ARM64ConditionCode cc = isCmpl ? TR::CC_GT : TR::CC_MI;
+
+   generateSrc2Instruction(cg, cmpOp, node, src1Reg, src2Reg); // compare
+   generateTrg1ImmInstruction(cg, TR::InstOpCode::movzx, node, trgReg, 0);
+   generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, doneLabel, TR::CC_EQ);
+   if (isCmpl)
+      generateTrg1ImmInstruction(cg, TR::InstOpCode::movzx, node, trgReg, 1);
+   else
+      generateTrg1ImmInstruction(cg, TR::InstOpCode::movnx, node, trgReg, 0); // -1
```

Comment:
I can't help but think these two generate functions can be combined into one to save some code space:
```
generateTrg1ImmInstruction(cg, isCmpl ? TR::InstOpCode::movzx : TR::InstOpCode::movnx , node, trgReg, isCmpl ? 1 : 0);
```

Reply:
I changed the code based on the suggestion.